wbooks/src/app/screens/dashboard/index.js

```diff
@@ -1,57 +1,64 @@
 import React, {Component} from 'react';
+import { connect } from 'react-redux';
 import Catalog from './components/Catalog';
 import Search from './components/Search';
-import * as bookService from '../../../services/bookService';
 import UnauthorizedPage from '../../components/UnauthorizedPage';
+import { actionCreators } from '../../redux';
 
 class Dashboard extends Component {
-  state = {
-    filteredBooks: [],
-    books: []
-  };
 
   componentWillMount(){
-    this.getBooks();
+    this.props.getBooks();
   }
 
-  getBooks = () =>{
-    this.setState({loading: true});
-    bookService.getBooks(this.onGetBooksSuccess, this.onGetBooksFailure);
-  }
-
-  onGetBooksSuccess = (books) => {
-    this.setState({loading: false, books: books, filteredBooks: books});
-  }
-  onGetBooksFailure = (error) => {
-    this.setState({loading: false, notAuthorized: true});
-  }
-
-  updateFilters = (dropdownFilter, textFilter) => {
-
-    if(!textFilter && !dropdownFilter) {
-      this.setState({filteredBooks:this.state.books});
+  getFilteredBooks = () => {
+    const titleFilter = this.props.titleFilter;
+    const genreFilter = this.props.genreFilter;
+    if(!titleFilter && !genreFilter) {
+      return this.props.books;
     }
     else {
-      const newBooks = this.state.books.filter((book) => (
-          (!textFilter || book.title.toLowerCase().includes(textFilter.toLowerCase()))
-          && (!dropdownFilter || book.genre.toLowerCase() === dropdownFilter.toLowerCase())
+      const newBooks = this.props.books.filter((book) => (
+          (!titleFilter || book.title.toLowerCase().includes(titleFilter.toLowerCase()))
+          && (!genreFilter || book.genre.toLowerCase() === genreFilter.toLowerCase())
           ));
-      this.setState({filteredBooks: newBooks});
-    }
+      return newBooks;
+    }    
   }
 
   render(){
-    if(this.state.loading)
+    if(this.props.booksLoading)
       return <h1>Cargando...</h1>
-    if(this.state.notAuthorized)
+    if(this.props.notAuthorized || this.props.booksFailed)
       return <UnauthorizedPage/>
+    const books = this.getFilteredBooks();
     return(
       <div className="dashboard">
-        <Search updateFilters={this.updateFilters}/>
-        <Catalog books={this.state.filteredBooks}/>
+        <Search/>
+        {
+          books.length > 0 ? 
+          <Catalog books={books}/> : 
+        <p className="no-books-found" style={{margin: 20}}>No se encontraron coincidencias</p>
```

Comment:
why margin by styles inLine? you have the no-books-found class

Reply:
um yes, i did that because i didnt have a style.css file and i would've had to create it only for that "margin: 20". But nontheless i think you're right, i should do it :stuck_out_tongue: .

Reply:
:+1: